static/_posts/2024-04-01-vgaterm-intro-001.md

```diff
@@ -0,0 +1,171 @@
+
+### Displaying Pixels
+To display pixels, the `CLK_VIS` signal generated by the VGA timing hardware reads bytes from the FIFO and into a [Programmable Logic Device (PLD)](https://en.wikipedia.org/wiki/Programmable_logic_device) which maps the 8 bits into 9 bits, 3 bits for each of the color channels - red, green, and blue. These 9 bits then get sent to the DAC, described in the following section.
+
+An important consideration for this stage is how long it takes to shuttle a byte from the FIFO into the DAC and finally into the monitor. Since there is processing of the read bytes it's possible that the pixels become late compared to `H_SYNC` and `V_SYNC` and this will show up as a horizontal shift. I.E. the "first" pixel could actually show up on the 2nd or 3rd pixel position on the screen, instead of the actual 1st pixel.
+
+{% include figure.html image="/assets/images/left-collum-timing-bug.jpg" caption="Notice the far left column of the screen in white next to the blue. It appears to wrap around from the far right side of the screen because the first pixel of the image is shifted over 2 or 3 positions" width=550 %}
+
+So we developed a method to handle this kind of pixel shifting to synchronize the pixel hardware with the VGA timing hardware.
+
+### Digital to Analog Converter (DAC)
+
+{% include figure.html image="/assets/images/vgaterm-dac-board.jpg" caption="R2R Resistor DAC" width=550 %}
+
+Once the pixel byte has been taken out of the FIFO memory it's on its way to being displayed by the monitor. To be displayed this pixel byte must be converted into three analog voltages between 0 and 0.7 volts corresponding to the Red, Green, and Blue color channels. We accomplished this with an [R2R Resistor Ladder DAC](https://en.wikipedia.org/wiki/Resistor_ladder#R%E2%80%932R_resistor_ladder_network_(digital_to_analog_conversion)).
+
+In an R2R resistor ladder DAC the resistor network is commonly connected to an op-amp to maintain the output voltage regardless of what load the DAC is connected to. However in our case the load is only ever the 75 Ω impedance of the analog color lines in the monitor, which we can account for in the resistor ladder design. One concern, however, is that our resistor ladder can consume a fair amount of current because we're going from 5 V input and dropping to under 1 V. The output pins representing the color coming from an IC may not be rated to supply the amount of current the DAC would need to maintain the voltage.
+
+In an earlier version of the DAC this is what we did (copying Ben Eater's video) but the colors will dull. We also tried an op-amp but that we saw some weird visual artifacts in the output. What seemed to work best was using a [line driver](https://www.ti.com/lit/gpn/SN74AHCT1G125) to supply the resister ladder with enough current to prevent the voltage dropping too low. Now the colors are properly vivid.
```

Comment:
```suggestion
In an earlier version of the DAC this is what we did anyway (copying Ben Eater's video), making the colors dull. We also tried an op-amp but we saw some weird visual artifacts in the output. What seemed to work best was using a [line driver](https://www.ti.com/lit/gpn/SN74AHCT1G125) to supply the resister ladder with enough current to prevent the voltage dropping too low. Now the colors are properly vivid.
```